### PR TITLE
Add position marker to some format strings

### PR DESCRIPTION
**Pull request by Qrox**
#### Summary
SUMMARY: I18N "Add position marker to some format strings"

#### Purpose of change
There is the need to reorder format parameters in some of the game's strings during translation, but it's currently impossible to do so on Transifex. While the parameters *can* be reordered, the resulting translation does not have proper position marker added.

#### Describe the solution
I searched the game's codebase using regex to find strings containing phrases in the form `<format specifier> <preposition> <word> <format specifier>`, where the two format specifier usually need to be swapped during translation (at least in Chinese), and changed the strings I deemed necessary.

#### Describe alternatives you've considered
I have filed a report to Transifex about the problem, but before they eventually fix it this will do for now.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -3443,16 +3444,19 @@ std::string item::tname( unsigned int quantity, bool with_prefix, unsigned int t
...
+            //~ %1$s: item name, %2$s: non-liquid, non-food, non-drink content item name
+            maintext = string_format( pgettext( "item name", "%1$s with %2$s" ), label( quantity ),
                                       contents_item.tname( quantity, with_prefix ) );
         }
     } else if( !contents.empty() ) {
+        //~ %1$s: item name, %2$zd: content size
         maintext = string_format( npgettext( "item name",
-                                             "%s with %zd item",
-                                             "%s with %zd items", contents.size() ),
+                                             "%1$s with %2$zd item",
```
Comment (AMurkin) [inline src/item.cpp]: A non-adjacent comment will not be extracted for this translation.

File: src/iuse_actor.cpp [hunk 2]
```diff
@@ -3708,7 +3710,9 @@ hp_part heal_actor::use_healing_item( player &healer, player &patient, item &it,
     } else {
         // Player healing NPC
         // TODO: Remove this hack, allow using activities on NPCs
-        const std::string menu_header = string_format( _( "Select a body part of %s for %s:" ),
+        //~ %1$s: patient name, %2$s: healing item name
+        const std::string menu_header = string_format( pgettext( "healing",
+                                        "Select a body part of %1$s for %2$s:" ),
```
Comment (AMurkin) [inline src/iuse_actor.cpp]: A non-adjacent comment will not be extracted for this translation.

File: src/item.cpp (hunk 1 above)
Comment (Qrox) [reply src/item.cpp]: I think it checks for the npgettext call instead of the actual string?

Comment (AMurkin) [reply src/item.cpp]: It doesn't seem like that.

Updated pot fragment for the line in my next comment:
```pot

#: src/iuse_actor.cpp
#, c-format
msgctxt "healing"
msgid "Select a body part of %1$s for %2$s:"
msgstr ""
```

**Comment by cainiaowu** [comment]
May have fixed #17787 .

**Pull request merged**